ssl/ssl_sess.c

```diff
@@ -452,6 +452,70 @@ int ssl_get_new_session(SSL *s, int session)
     return 1;
 }
 
+SSL_SESSION *lookup_sess_in_cache(SSL *s, const unsigned char *sess_id,
+                                  size_t sess_id_len)
+{
+    SSL_SESSION *ret = NULL;
+    int discard;
+
+    if ((s->session_ctx->session_cache_mode
+         & SSL_SESS_CACHE_NO_INTERNAL_LOOKUP) == 0) {
+        SSL_SESSION data;
+
+        data.ssl_version = s->version;
+        memcpy(data.session_id, sess_id, sess_id_len);
```

Comment:
Would a length check be prudent here?  Are we sure the caller's `sess_id_len` can never be too large?

Reply:
This is an internal function only and the callers should only ever be sending in sane session id lengths - so this is a "should not happen" scenario. However, to be on the safe side, I added an `ossl_assert` check.